app/src/main/java/com/example/sharingang/utils/notification/MyFirebaseMessagingService.kt

```diff
@@ -21,9 +23,22 @@ class MyFirebaseMessagingService: FirebaseMessagingService() {
         remoteMessage.data.let {
             val userId = remoteMessage.data["userId"]
             val deeplink = remoteMessage.data["deeplink"]
-            if (userId != currentUserProvider.getCurrentUserId() && deeplink != null) {
+            var channelId = ""
+            val title = when (remoteMessage.data["notificationType"]) {
+                "new_item" -> {
+                    channelId = NotificationFields.NEW_ITEM_CHANNEL_ID
+                    applicationContext.getString(R.string.new_item_notification_message)
+                }
+                "chat" -> {
+                    channelId = NotificationFields.CHAT_CHANNEL_ID
+                    remoteMessage.data["fromName"]+":"
+                }
+                else -> ""
+            }
+            val to = remoteMessage.data["toId"]
+            if ((userId != null && userId != currentUserProvider.getCurrentUserId() || (to != null && to == currentUserProvider.getCurrentUserId())) && deeplink != null) {
```

Comment:
Uh, can we simplify this condition, or extract out some meaningful subconditions?

Reply:
You're right, I'll correct that right away!